app/src/main/java/nl/tudelft/pixelperfect/event/parameter/EventParameter.java

```diff
@@ -0,0 +1,152 @@
+package nl.tudelft.pixelperfect.event.parameter;
+
+import com.jme3.network.serializing.Serializable;
+
+/**
+ * Class representing a single parameter of an Event, having either a discrete value
+ * (EventParameterValue) or a continuous one (Integer).
+ *
+ * @author Jesse Tilro
+ *
+ */
+@Serializable
+public class EventParameter {
+
+    private String key;
+    private String summary;
+    private EventParameterValues value;
+    private int numberValue;
+
+    /**
+     * Construct an EventParameter with a specific value (part of the EventParameterValue
+     * enumeration).
+     *
+     * @param key
+     *          The key of the parameter.
+     * @param value
+     *          The specific value of the parameter.
+     */
+    public EventParameter(String key, EventParameterValues value) {
+        this.key = key;
+        this.value = value;
+        this.numberValue = 0;
+        this.summary = "";
+    }
+
+    /**
+     * Construct an EventParameter with a generic value (a number).
+     *
+     * @param key
+     *          The key of the parameter.
+     * @param numberValue
+     *          The number value of the parameter.
+     */
+    public EventParameter(String key, int numberValue) {
+        this.key = key;
+        this.value = EventParameterValues.GENERIC;
+        this.numberValue = numberValue;
+        this.summary = "";
+    }
+
+    /**
+     * Get this parameter's key.
+     *
+     * @return This parameter's key.
+     */
+    public String getKey() {
+        return key;
+    }
+
+    /**
+     * Check whether this parameter has a continuous value.
+     *
+     * @return A boolean.
+     */
+    public boolean isGeneric() {
+        return (value == EventParameterValues.GENERIC);
+    }
+
+    /**
+     * Get this parameter's number value.
+     *
+     * @return A number value.
+     */
+    protected int getNumberValue() {
+        return numberValue;
+    }
+
+    /**
+     * Get this parameter's specific value.
+     *
+     * @return A specific value.
+     */
+    protected EventParameterValues getValue() {
```

Comment:
The class is already named Parameter, so `EventParameter.getValue` should already imply "get the parameter's value". Why would you like to name this to `EventParameter.getParameterValue`? (In my opinion it would be as redundant as `Spaceship.getSpaceshipEventLog` or `Game.getGameSpaceship` or `Event.getEventSummary`.